src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.ts

```diff
@@ -297,6 +337,27 @@ export class ProgrammingExerciseUpdateComponent implements OnInit {
         this.programmingExercise.title = undefined;
     }
 
+    /**
+     * Sets the attribute invalidRepositoryNamePattern to an updated RegExp that does not allow auxiliary repository names that are already used for this exercise and only allows
+     * "-" besides [A-z]
+     */
+    private setInvalidRepoNamePattern() {
+        let invalidRepoNames = '';
+        this.programmingExercise.auxiliaryRepositories?.forEach((auxiliaryRepository) => (invalidRepoNames += '|' + auxiliaryRepository.name));
+        this.invalidRepositoryNamePattern = new RegExp('^(?!(solution|exercise|tests' + invalidRepoNames + ')\\b)\\b(\\w|-)+$');
+    }
+
+    /**
+     * Sets the attribute invalidDirectoryNamePattern to an updated RegExp that does not allow directory names that are already used for other auxiliary repositories of this
+     * exercise "-" besides [A-z]
+     */
+    private setInvalidDirectoryNamePattern() {
+        let invalidDirectoryNames = '';
+        this.programmingExercise.auxiliaryRepositories?.forEach((auxiliaryRepository) => (invalidDirectoryNames += '|' + auxiliaryRepository.checkoutDirectory));
+        this.invalidDirectoryNamePattern = new RegExp('^(?!( ' + invalidDirectoryNames + ')\\b)\\b(\\w|-|/)+$');
+        console.log('Invalid Directory Name pattern: ' + this.invalidDirectoryNamePattern);
```

Comment:
To be removed

Reply:
Fixed: https://github.com/ls1intum/Artemis/pull/3457/commits/5d2ec4392aeebaab32d45cb91bd6b9373737e7c6